CMakeLists.txt

```diff
@@ -0,0 +1,62 @@
+cmake_minimum_required(VERSION 3.14)
+
+project(pennylane_lightning)
+
+# Read and set pennylane_lightning version
+execute_process(COMMAND "/usr/bin/env" "python"  "-c"
+	"with open('${CMAKE_SOURCE_DIR}/pennylane_lightning/_version.py') as f: print(f.readlines()[-1].split()[-1].strip(\"\\\"'\"), end='')"
+	OUTPUT_VARIABLE VERSION_STRING)
+message(STATUS "Pennylane_lightning version ${VERSION_STRING}")
+set(PROJECT_VERSION ${VERSION_STRING})
+
+set(CMAKE_CXX_STANDARD 11) # At least C++11 is required
+find_package(OpenMP REQUIRED) # find OpenMP
+
+if(NOT CMAKE_BUILD_TYPE)
+	set(CMAKE_BUILD_TYPE Release)
+endif()
+
+set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -march=native -W -DVERSION_INFO=${VERSION_STRING}")
```

Comment:
It may be a good idea to split the native compilation into a user-selectable option, such as:
`option(ENABLE_NATIVE "Enable native CPU build tuning" OFF)` which can then be enabled per target using [target_compile_options](https://cmake.org/cmake/help/v3.14/command/target_compile_options.html) with something like 
```
if(ENABLE_NATIVE)
    target_compile_options(TargetName INTERFACE -march=native)
endif()
```
This way we can more easily see the performance differences with the library depending on architecture-specific features.

Reply:
Thanks! I incorporate this in f0066d3d417c01c0f0f716851f75fc7447dd5ac8.